packages/parse5/lib/parser/index.ts

```diff
@@ -82,7 +82,7 @@ export interface ParserOptions<T extends TreeAdapterTypeMap> {
      *
      *  @default `true`
      */
-    scriptingEnabled?: boolean | undefined;
+    scriptingEnabled?: boolean;
```

Comment:
Is it intentional that the value can’t be `undefined`, but that the field can be missing?

Reply:
These values can still be explicitly undefined — I am honestly not sure why the `| undefined` was here in the first place (this was copied from the previous @types package). My suspicion would be an older TS version?

The only difference is that now `Required<>` works with the options.

Reply:
I know that DefinitelyTyped was adding explicit `| undefined` to types, when they had `?`. They‘re planning to pull them apart: https://github.com/DefinitelyTyped/DefinitelyTyped/commit/b24e5f670af4b29aa131dedc1db578d08d40c028

Reply:
Ahh, thanks for digging this up. Looks like the option is now called `exactOptionalPropertyTypes` and is explicitly opt-in (no longer enabled with `strict`): https://github.com/microsoft/TypeScript/pull/44626

My intuition is that users enabling this flag wouldn't want to be able to pass `undefined` here.